libgadget/powerspectrum.c

```diff
@@ -78,8 +76,8 @@ void powerspectrum_sum(struct _powerspectrum * PowerSpectrum, const double BoxSi
         PowerSpectrum->Power[i] /= PowerSpectrum->Norm;
         PowerSpectrum->kk[i] /= PowerSpectrum->Nmodes[i];
         /* Mpc/h units */
-        PowerSpectrum->kk[i] *= 2 * M_PI / (BoxSize_in_cm / CM_PER_MPC );
-        PowerSpectrum->Power[i] *= pow(BoxSize_in_cm / CM_PER_MPC , 3.0);
+        PowerSpectrum->kk[i] *= 2 * M_PI / (PowerSpectrum->BoxSize_in_cm / CM_PER_MPC );
+        PowerSpectrum->Power[i] *= pow(PowerSpectrum->BoxSize_in_cm / CM_PER_MPC , 3.0);
```

Comment:
Perhaps BoxSize_in_MPC is cleaner than cm.

Reply:
Indeed!